frontend/apps/filemanager/filemanagerutil.lua

```diff
@@ -59,8 +63,100 @@ function filemanagerutil.resetDocumentSettings(file)
             end
         end
         doc_settings:makeTrue("docsettings_reset_done") -- for readertypeset block_rendering_mode
-        doc_settings:close()
+        doc_settings:flush()
     end
 end
 
+-- Get a document's status ("new", "reading", "complete", or "abandoned")
+function filemanagerutil.getStatus(file)
+    local status = "new"
+    if DocSettings:hasSidecarFile(file) then
+        local docinfo = DocSettings:open(file) -- no io handles created, do not close
+        if docinfo.data.summary and docinfo.data.summary.status and docinfo.data.summary.status ~= "" then
+            status = docinfo.data.summary.status
+        else
+            status = "reading"
+        end
+    end
+    return status
+end
+
+-- Set a document's status
+function filemanagerutil.setStatus(file, status)
+    -- In case the book doesn't have a sidecar file, this'll create it
+    local docinfo = DocSettings:open(file)
+    local summary
+    if docinfo.data.summary and docinfo.data.summary.status then
+        -- Book already had the full BookStatus table in its sidecar, easy peasy!
+        docinfo.data.summary.status = status
+        docinfo.data.summary.modified = os.date("%Y-%m-%d", os.time())
+        summary = docinfo.data.summary
+    else
+        -- No BookStatus table, create a minimal one...
+        if docinfo.data.summary then
+            -- Err, a summary table with no status entry? Should never happen...
+            summary = { status = status }
+            -- Append the status entry to the existing summary...
+            require("util").tableMerge(docinfo.data.summary, summary)
+            docinfo.data.summary.modified = os.date("%Y-%m-%d", os.time())
+            summary = docinfo.data.summary
+        else
+            -- No summary table at all, create a minimal one
+            summary = {
+                status = status,
+                modified = os.date("%Y-%m-%d", os.time())
+            }
+        end
+    end
+    docinfo:saveSetting("summary", summary)
+    docinfo:flush()
+end
+
+-- Generate all book status file dialog buttons in a row
+function filemanagerutil.getStatusButtonsRow(file, caller_callback)
```

Comment:
Maybe `gen`? (generate)

Reply:
Hmm?

Reply:
I mean `genStatusButtonsRow`.

Reply:
Good eye, didn't even see it until looking at it 10 times